src/jansson.h

```diff
@@ -50,6 +52,7 @@ typedef enum {
 typedef struct json_t {
     json_type type;
     size_t refcount;
+    pthread_mutex_t refmutex;
```

Comment:
This struct is probably [pretty big](http://refspecs.linuxbase.org/LSB_3.1.0/LSB-Core-generic/LSB-Core-generic/libpthread-ddefs.html), but I don't see a good way around it.